From: Rafael J. Wysocki <rjw@suse.de>
Subject: x86: Fix unknown_nmi_error()
References: bnc#593731
Patch-mainline: no

Patch patches.suse/x86-add-kdb-support-for-unknown_nmi_error-handler.patch
changed the behavior of unknown_nmi_error() in such a way that it
doesn't work as expected on some system.  Fix this by making
the second half of unknown_nmi_error() build (and consequently
execute) also in the CONFIG_KDB case. 

Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/kernel/traps.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

Index: linux-2.6.32-SLE11-SP1/arch/x86/kernel/traps.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/arch/x86/kernel/traps.c
+++ linux-2.6.32-SLE11-SP1/arch/x86/kernel/traps.c
@@ -380,8 +380,7 @@ unknown_nmi_error(unsigned char reason, 
 		spin_unlock_irqrestore(&kdb_nmi_lock, flags);
 		(void)kdb(KDB_REASON_ENTER_SLAVE, reason, regs);
 	}
-	return;
-#else
+#endif /* CONFIG_KDB */
 
 	if (notify_die(DIE_NMIUNKNOWN, "nmi", regs, reason, 2, SIGINT) ==
 			NOTIFY_STOP)
@@ -405,7 +404,6 @@ unknown_nmi_error(unsigned char reason, 
 		panic("NMI: Not continuing");
 
 	printk(KERN_EMERG "Dazed and confused, but trying to continue\n");
-#endif /* CONFIG_KDB */
 }
 
 static notrace __kprobes void default_do_nmi(struct pt_regs *regs)
